public/js/controllers/address.js

```diff
@@ -117,6 +117,21 @@
                     visibility: [true],
                     fillGraph: true
                 }
+
+                _this.defaultHash = 'list-view'
+                var hashVal = window.location.hash.replace('#', '') || _this.defaultHash;
+
+                if (hashVal.length === 0 || hashVal === _this.defaultHash){
+                    history.pushState({},  this.addr, "#" + _this.defaultHash);
+                } else {
+                    var selectedVal = this.optionsTarget.namedItem(hashVal)
+                    $(this.optionsTarget).val((selectedVal ? selectedVal.value : 'types'))
+
+                    $('#addr-btn').removeClass('btn-active');
```

Comment:
If you goto #types and try sharing that url. The person who will open that link will see the list view with both buttons styled as active.
<img width="147" alt="btns bug" src="https://user-images.githubusercontent.com/25571523/48050950-76204c80-e158-11e8-9aee-16ce82b5b8d8.png">

This should be addressed since the whole point of url backed state is to make views reproducible on page load.

It looks like the cause of the bug is a this selector should be ".addr-btn". You could simply change that "#" to a "."

...But do you want to take a shot a switching those to stimulus targets instead?  The zoomOptionTargets in charts.js are pretty similar for reference. 

I know it's a bit more verbose, but it leads to more robust javascript and also better collaboration across contributors as the next person can quickly understand how things work by simply glancing at the template markup and seeing which actions are hooked up to what elements.

Reply:
@dmigwi Most regular JS event callbacks (i.e. click, hover) have been implemented using a _stimulus.js_ controller. I'm wasn't familiar with _stimulus_, and there is a bit of a learning curve, but you could take a look over [the docs](https://stimulusjs.org/reference/controllers), and see how some of the controllers have been set up for dcrdata in `/public/js/controllers/`.